### ISSUE TITLE
Deprecate twisted.spread.refpath

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3723]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3723|
|Type|enhancement|
|Created|2009-03-30 21:40:52Z|

This module's purpose is not really coherent and as a somewhat sophisticated PB user I have no idea what I would want to use it for, nor have ever used it.  Let's get rid of it ASAP.

Attachments:

* [deprecate-refpath.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/be9/be98fef84ca4e0d638f2511d975ba3962fdaecdb/1383acdba343fba6c25c4c0eebc3be4aeb0903e8.patch) (986 bytes) - added by exarkun on 2009-04-11 22:32:04Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3723 3723
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1238449252000000 1238449252000000
changetime__1239655348000000 1239655348000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

+1

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Attached patch adds a module level deprecation.  No unit tests because I still don't know how to unit test this kind of deprecation.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Alright, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [26694]) Apply deprecate-refpath.patch

Author: exarkun
Reviewer: therve
Fixes: [#3723](https://github.com/twisted/twisted/issues/3723)

Deprecate `twisted.spread.refpath`, an untested, very minimally documented module
which implements a more complicated reference scheme for PB.  This is functionality
which almost no one ever needs; it should be re-developed test-driven if any one
does, in any case.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
